### PR TITLE
Define type for plan payment service

### PR DESCRIPTION
**Pull request by loan9202**
### Purpose
- Define type for plan payment service

### Pull Request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

### How to test ?
1. Click to setting app
2. Click tab `Choose a plan`
3. Update Plan information
4. Click Next -> Subscribe or Save Button

### What to check ?
- See the chang on current plan

### CODE REVIEW

File: lib/services/plan_payment_service.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  coupon?: string;
+
+  /** The payment method on stripe */
+  paymentMethodId: string;
+
+  /** The ID of the application. */
+  appId?: string;
+
+  /** The check option plan for user by year or month. */
+  yearly?: boolean;
```
Comment (fouadmen) [inline lib/services/plan_payment_service.ts:14]: If attribute or variable is boolean use `is...`.
Also please be more specific.
```suggestion
  isYearlyPayment?: boolean;
```

File: lib/services/plan_payment_service.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Stripe } from 'stripe';
+
+export interface PlanPaymentRequestBody {
+  /** The promotion codes for coupon */
+  coupon?: string;
```
Comment (fouadmen) [inline lib/services/plan_payment_service.ts:5]:
```suggestion
  couponCode?: string;
```

File: lib/services/plan_payment_service.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  /** The payment method on stripe */
+  paymentMethodId: string;
```
Comment (fouadmen) [inline lib/services/plan_payment_service.ts:8]:
```suggestion
  stripePaymentMethodId: string;
```

File: lib/services/plan_payment_service.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  /** The ID of the application. */
+  appId?: string;
+
+  /** The check option plan for user by year or month. */
+  yearly?: boolean;
+
+  /** The ID of list-plan. */
+  planSelected: string;
+}
```
Comment (fouadmen) [inline lib/services/plan_payment_service.ts:18]:
```suggestion
  /** The ID of the selected payment plan */
  selectedPlanId: string;
}
```

File: lib/services/plan_payment_service.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+export type PlanPaymentResponse = Stripe.Response<Stripe.Subscription> & {
+  /** tottal resquest list plan */
+  totalRequest: number;
```
Comment (fouadmen) [inline lib/services/plan_payment_service.ts:22]: Please be careful about typos
```suggestion
  /** The total number of requests for the payment plan */
  totalRequests: number;
```

File: lib/services/plan_payment_service.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { Stripe } from 'stripe';
+
+export interface PlanPaymentRequestBody {
```
Comment (fouadmen) [inline lib/services/plan_payment_service.ts:3]: Please follow the naming conventions on click types repo.
```suggestion
export interface CreatePlanPaymentRequestBody {
```

**Review by fouadmen — CHANGES_REQUESTED** [review]
@loan9202 Looks good, but please be careful about the naming, typos and be sure to follow the naming conventions of click types ;)
Thank you!

File: lib/services/plan_payment_service.ts (hunk 1 above)
Comment (loan9202) [reply lib/services/plan_payment_service.ts]: I updated

File: lib/services/plan_payment_service.ts (hunk 3 above)
Comment (loan9202) [reply lib/services/plan_payment_service.ts]: I updated

File: lib/services/plan_payment_service.ts (hunk 5 above)
Comment (loan9202) [reply lib/services/plan_payment_service.ts:22]: I updated

File: lib/services/plan_payment_service.ts (hunk 6 above)
Comment (loan9202) [reply lib/services/plan_payment_service.ts]: I updated

File: lib/services/plan_payment_service.ts (hunk 2 above)
Comment (loan9202) [reply lib/services/plan_payment_service.ts]: I updated

**Review by fouadmen — APPROVED** [review]
LGTM! thanks!

File: lib/services/plan_payment_service.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  appId?: string;
+
+  /** The check option plan for user by year or month. */
+  isYearlyPayment?: boolean;
+
+  /** The ID of list-plan. */
+  selectedPlanId: string;
+}
+
+export type PlanPaymentResponse = Stripe.Response<Stripe.Subscription> & {
```
Comment (fouadmen) [inline lib/services/plan_payment_service.ts:20]: @loan9202 
I missed this one, please be careful with the naming.
```suggestion
export type CreatePlanPaymentResponse = Stripe.Response<Stripe.Subscription> & {
```

**Review by fouadmen — CHANGES_REQUESTED** [review]
I noticed this return type, please be careful with the naming, thank you !

File: lib/services/plan_payment_service.ts (hunk 7 above)
Comment (loan9202) [reply lib/services/plan_payment_service.ts]: I updated

**Review by fouadmen — APPROVED** [review]
looks good thanks 👍

**Pull request merged**